pyerrors/fits.py

```diff
@@ -151,6 +151,12 @@ def func_b(a, x):
         For details about how the covariance matrix is estimated see `pyerrors.obs.covariance`.
         In practice the correlation matrix is Cholesky decomposed and inverted (instead of the covariance matrix).
         This procedure should be numerically more stable as the correlation matrix is typically better conditioned (Jacobi preconditioning).
+    corr_matrix: array, optional
+        if correlated_fit=True is set as well, can provide a correlation matrix (without y errros!) of your own choosing for a correlated fit
```

Comment:
I would exactly specify what is needed here, i.e., a square matrix of floats with the dimension N where N is the length of 'y'. Once could also write that this is the correlation matrix of the Observables in `y`.

Reply:
Yes, that's a good idea! It even has to be positive definite otherwise it's not invertible and the Cholesky decomposition is also not applicable, right? So the way the correlation matrix is constructed by
the `covariance` function in pyerrors (used in the `least squares`  for a correlated fit without any cov/corr matrix input)  only guarantees  a positive semi-definite matrix so you can run into a problem then too 
or am I missing something? Of course the condition number of the correlation matrix can also just be bad even if the matrix is positive definite (and therefore in principle invertible). In both cases the code
produces an error but wouldn't it still be good to also stress that these two conditions (positive definite and well-conditioned) have to be met by the correlation matrix handed over as `corr_matrix`?

Reply:
That's true. It has to be a symmetrix, positive definite matrix. The condition number is an implicit problem, I would say.